tests/integration/main-test.js

```diff
@@ -0,0 +1,177 @@
+import { module, test } from 'qunit';
+import { visit, currentURL, click, fillIn, pauseTest, find, findAll } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import { currentSession, authenticateSession, invalidateSession } from 'ember-simple-auth/test-support';
+import { tracked } from '@glimmer/tracking';
+import Service from '@ember/service';
+import Adapter from 'ember-local-storage/adapters/local';
+import resetStorages from 'ember-local-storage/test-support/reset-storage';
+
+module('Acceptance | Main flow of Application', function(hooks) {
+  setupApplicationTest(hooks);
+  
+  //Preparation before test
+  hooks.beforeEach(function() {
+    //This service is needed to mock the real service that needs a real conection with firebase auth
+    //to retrieve logged user email
+    this.owner.register('service:logged-user', class TestService extends Service {
+      @tracked user = {email: 'user@teste.com'}
+    });
+
+    //This adapter is used to use ember local storage instead of firestore
+    this.owner.register('adapter:application', class TestService extends Adapter {
+   
+    });
+  });
+
+  //This makes the database empty after the end of the test
+  hooks.afterEach(function() {
+    if (window.localStorage) {
+      window.localStorage.clear();
+    }
+    if (window.sessionStorage) {
+      window.sessionStorage.clear();
+    }
+    resetStorages();
+  });
+
+  test('Main flow of the application', async function(assert) {
+
+    //These three asserts prove that an unauthorized user can´t see the content of the pages and are redirected
+    //to login form
+    await visit('admin/invitations');
+    assert.equal(currentURL(), '/login', 'Assert unauthorized route admin/invitations');
+
+    await visit('admin/contacts');
+    assert.equal(currentURL(), '/login', 'Assert unauthorized route admin/contacts');
+
+    await visit('admin/seeder');
+    assert.equal(currentURL(), '/login', 'Assert unauthorized route admin/seeder');
+
+    //Sending an invitation. Also check button is disabled with no e-mail written and not disable with valid
+    //email
+    await visit('');
+    assert.dom('[data-test="saveInvitation"]').isDisabled();
+    await fillIn('[data-test="emailField"]', 'cool@cool.com');
```

Comment:
@tpetrone  I didn´t get it. What do you mean by hardcode the email? I can run the test a lot of times and it will always pass. Remember that the database is reset anytime you do a test. Here I put the email and a little bit down I assert that the email is recorded in the database in the admin/invitations route. You mean by just creating a variable let email = test@test.com and filling the form using email variable?

Reply:
Just to complement. Your comment is very cloudy to me. I´m filling the email field of the "/" route where you are supposed to write a valid email so it saves your email in the database....

Reply:
And the only way to test with an invalid email is with isDisabled an isNotDisabled because the button will be only able with a valid email.....

Reply:
@tpetrone I cannot get the point of it. The test will always pass! if I change the email here the form will be sended and the assert of the email in the table (admin/invitations) will fail......